src/main/scala/com/github/sbt/git/GitPlugin.scala

```diff
@@ -112,8 +112,8 @@ object SbtGit {
   // Build settings.
   import GitKeys._
   def buildSettings = Seq(
-    useConsoleForROGit := false,
-    gitReader := new DefaultReadableGit(baseDirectory.value, if (useConsoleForROGit.value) Some(new ConsoleGitReadableOnly(ConsoleGitRunner, file("."), sLog.value)) else None),
+    useConsoleForROGit := isGitLinkedRepo(baseDirectory.value),
```

Comment:
It would be nice if we could check if jgit v7 is used and if so make `useConsoleForROGit` false, no matter if we are in a linked repo or not (because jgit 7 supports worktrees, read). Like:
```sbt
    useConsoleForROGit := jGitVersionLtOrEq6 && isGitLinkedRepo(baseDirectory.value),
```

I looked into jgit but they do not publish the version in a variable nor method... So I don't see a way of doing this easily.

Reply:
On Maven Central [we can find the latest jgit v6 and first v7 version published](https://repo1.maven.org/maven2/org/eclipse/jgit/org.eclipse.jgit/maven-metadata.xml). [The diff is quite big](https://github.com/eclipse-jgit/jgit/compare/v6.10.0.202406032230-r...v7.0.0.202409031743-r), most of the things we don't need to look at.
So I provide [my own html rendered diff here](https://mkurz.github.io/jgit_v6.10.0.202406032230-r..v7.0.0.202409031743-r.html) from this command:

```
git diff v6.10.0.202406032230-r..v7.0.0.202409031743-r org.eclipse.jgit
```

[You can see in v7 the class `org.eclipse.jgit.lib.Constants` added the `GITDIR_FILE` constant](https://mkurz.github.io/jgit_v6.10.0.202406032230-r..v7.0.0.202409031743-r.html#d2h-701406) which is directly related to git worktree.

Would it be able to make use of reflection and check if the `Contants` class contains the field `GITDIR_FILE` and if so, we can treat the jgit version on the classpath as one that supports git worktree?
Like:
```sbt
    useConsoleForROGit := !jGitWithWorktreeSupportDetected() && isGitLinkedRepo(baseDirectory.value),
```
and in the `jGitWithWorktreeSupportDetected` method put the reflection code?

What do you think?

Reply:
Stepping back a bit, instead of making the automatic detection more complicated, would it make sense to make this more of a machine-wide setting that someone could always opt out of JGit? Also in general, I don't really get the motivation behind JGit. Is the assumption that someone could be working on an opensource project but somehow do not have `git` installed on their PATH?

Reply:
See also sbt-pgp. In https://github.com/sbt/sbt-pgp/pull/146, I defaulted to using command line `gpg` and I think the plugin became significantly more reliable because it relied less on the never-ending emulation of gpg via Bouncy Castle.